node/sc/bridgepool/sorted_map_list.go

```diff
@@ -100,6 +100,23 @@ func (m *ItemSortedMap) Put(event itemWithNonce) {
 	m.mu.Lock()
 	defer m.mu.Unlock()
 
+	m.put(event)
+}
+
+// PutWithLimit inserts a new item into the map with size limit of the map.
+func (m *ItemSortedMap) PutWithLimit(event itemWithNonce, limit int) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	if len(m.items) >= limit {
+		return
```

Comment:
Does the caller need to know that the item insertion failed?

Reply:
For now, there is nothing to do for that case. Therefore I didn't add a routine to return the error.

Reply:
I added an error. :) Thank you so much.